cmd/mistryd/end_to_end_test.go

```diff
@@ -176,28 +176,57 @@ func TestResultCache(t *testing.T) {
 
 func TestResultCacheExitCode(t *testing.T) {
 	cmdout1, _, err := cliBuildJob("--json-result", "--project", "result-cache-exitcode")
-	if err == nil {
-		t.Fatal("Expected error")
+	if err != nil {
+		t.Fatal(err)
 	}
 	bi, err := parseClientJSON(cmdout1)
 	if err != nil {
 		t.Fatal(err)
 	}
-	assertEq(bi.ExitCode, 33, t)
+	assertEq(bi.ExitCode, 0, t)
 	assertEq(bi.Cached, false, t)
 
 	cmdout2, _, err := cliBuildJob("--json-result", "--project", "result-cache-exitcode")
-	if err == nil {
-		t.Fatal("Expected error")
+	if err != nil {
+		t.Fatal(err)
 	}
 	bi, err = parseClientJSON(cmdout2)
 	if err != nil {
 		t.Fatal(err)
 	}
-	assertEq(bi.ExitCode, 33, t)
+	assertEq(bi.ExitCode, 0, t)
 	assertEq(bi.Cached, true, t)
 }
 
+func TestRerunFailedBuild(t *testing.T) {
+	// schedule a build that fails (non zero exit code)
+	cmdout, _, err := cliBuildJob("--json-result", "--project", "exit-code")
+	if err == nil {
+		t.Fatal("expected error")
+	}
+
+	br, err := parseClientJSON(cmdout)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	assertNotEq(br.ExitCode, 0, t)
```

Comment:
Why the exit code is 0 here? Shouldn't it be 33?

Reply:
we're testing whether the result was cached. a result is now only cached with an exit code of 0. so the test case now needs an exit code of 0

Reply:
i see we have another test for that actually, so I'm going to remove this one